src/redux/selectors/locationsSelectors.js

```diff
@@ -0,0 +1,15 @@
+import { createSelectorCreator, defaultMemoize } from 'reselect';
+import isEqual from 'lodash.isequal';
+
+const locationsSelector = state => state.locations.locations;
+export const isLocationsLoadingSelector = state => state.locations.isLoading;
+
+// I realize this is a bit of overkill, just wanted to try out the selectors
+const createDeepEqualSelector = createSelectorCreator(defaultMemoize, isEqual);
+
+export const transformLocations = createDeepEqualSelector(locationsSelector, locations =>
+  locations?.reduce((acc, location) => {
```

Comment:
correct me if I'm wrong: are locations taken from the initial state of redux? if so  - then they should be defined as [] in a worst case scenario. 

also... is this a much better situation for map, rather than reduce? 

```
locations.map(location => ({...location, fullName: `${location.name}, ${location.country}` });
```

Reply:
They're null in the initial state, but I can fix that